### PR TITLE
delete swc.py

### PR DESCRIPTION
**Pull request by alishakodibagkar**
- Removed swc.py and test_swc.py
  - Made changes to these 11 files to implement Neuron_trace.py:
       - brainlit/utils/__init__.py
       - brainlit/utils/upload.py
       - brainlit/utils/session.py
       - experiments/ffn/generating_image_masks.ipynb
       - experiments/axon_geometry/scripts/generate_trace_data.py
       - tests/test_fit_spline.py
       - tests/test_tube_seg.py
       - brainlit/archive/upload_skeleton.py
       - docs/notebooks/utils/downloading_benchmarking.ipynb
       - docs/notebooks/visualization/neighborhood_visualization_demo.ipynb
       - docs/notebooks/utils/downloading_brains.ipynb
   - Added “get_sub_neuron_paths” to Neuron_trace.py since “get_sub_neuron” returns a subneuron in graph format
       - Added example of get_sub_neuron_paths in NeuronTrace_tutorial.ipynb
       - Added get_sub_neuron_paths to test_Neuron_trace.py, passed pytest
   - Changed “get_segments” in session.py to return in graphs form and paths form
       - Changed test_session.py for testing get_segments in session.py, since I now return the segments in graphs form and paths form
   - Next step that needs to be made: make sure notebooks are cleaned up (often times I didn't have the specific files/filepaths that people used for notebooks so I made sure the changes I made worked using my own paths for swc files or s3 files. This messed up notebook outputs so I need to go back in and try my best to replicate outputs)
   - Questions: 
       * What is [viz.rst](https://github.com/neurodata/brainlit/blob/develop/docs/reference/viz.rst), should I change it?
       * [loading.ipynb](https://github.com/neurodata/brainlit/blob/6410a3b613dabdaa15f0c149b51c7c512019c08e/docs/notebooks/visualization/loading.ipynb) mainly demonstrates swc.py functions. Should I still include this file? If so ,how should I go about editing it?
       * [Installation_on_win.md](https://github.com/neurodata/brainlit/blob/a117ac52efec2224a4311c9167b33d46d90b3f05/docs/Installation_on_Win7.md) shows “detail of code history”  for installation problem, so I didn’t remove swc.py functions from that text. Is that okay?
       * [tests/archive/test_visualize.py](https://github.com/neurodata/brainlit/blob/0544d6159a663ae26099e5ec2bc0aa80a20623af/tests/archive/test_visualize.py) was fully commented out. I assumed this means I don’t need to change it/it's not needed?

### CODE REVIEW

**Comment by alishakodibagkar** [comment]
-  Removed swc.py functions from viz.rst
-  Removed NeuronTrace_tutorial and added documentation to Neuron_trace.py instead
-  Fixed docs with links to data not working (added data to brainlit/data for one of the docs)
-  Updated loading.ipynb to demonstrate functions of Neuron_trace.py

**Review by tathey1 — CHANGES_REQUESTED** [review]
A few notes:
- I believe codecov only went down because you are removing a large file. Regardless, can you please increase coverage of NeuronTrace to 90%? You are almost there
- Is there not going to be a Neuron_Trace tutorial anymore? If so, can you please give reasoning here in this PR why you decided that?
- Remove notebook output, you can do this in jupyter notebook with "clear all output" or on the command line with nbstripout, described [here](https://github.com/neurodata/brainlit/issues/84)
- Please add an entry to [this file](https://github.com/neurodata/brainlit/blob/develop/docs/reference/utils.rst), where you just copy the entry for Neuroglancersession, except adapt it to your file, in order to get your docs rendered on our website.

**Comment by alishakodibagkar** [comment]
- Code coverage improved for Neuron_trace.py (covered above lines in test)
- I removed the Neuron_Trace tutorial and instead updated brainlit/docs/notebooks/visualization/loading.ipynb. This file previously demonstrated the use of functions from swc.py to load neuron data. I changed the file to demonstrate the use of functions from Neuron_trace.py
- Used nbstripout to remove notebook output for all notebooks under brainlit/docs/notebooks
- Updated brainlit/docs/reference/utils.rst

File: brainlit/utils/Neuron_trace.py [hunk 1]
```diff
@@ -43,6 +43,17 @@ class NeuronTrace:
             If s3 bucket path is provided, the segment number to pull
         mip : None,int
             If s3 bucket path is provided, the resolution to use for scaling
+
+    Example
+    ----------
+    >>> swc_path = "../../../../Manual-GT/8-01_test_1-5/8-01_test_1/tree_2.swc"
```
[the remark this answers is not in the document]
Comment (alishakodibagkar) [reply brainlit/utils/Neuron_trace.py]: changed this

File: brainlit/utils/Neuron_trace.py [hunk 2]
```diff
@@ -410,6 +532,70 @@ def get_sub_neuron(self, bounding_box, spacing=None, origin=None):
...
+        check_type(spacing, (type(None), np.ndarray))
+        if type(spacing) == np.ndarray:
+            check_size(spacing)
+        check_type(origin, (type(None), np.ndarray))
+        if type(origin) == np.ndarray:
+            check_size(origin)
+
+        # if origin isn't specified but spacing is, set origin to np.array([0, 0, 0])
+        if type(spacing) == np.ndarray and origin is None:
+            origin = np.array([0, 0, 0])
```
[the remark this answers is not in the document]
Comment (alishakodibagkar) [reply brainlit/utils/Neuron_trace.py:582]: code coverage no longer an issue

File: brainlit/utils/Neuron_trace.py [hunk 3]
```diff
@@ -410,6 +532,70 @@ def get_sub_neuron(self, bounding_box, spacing=None, origin=None):
...
+
+        # voxel conversion option
+        if type(spacing) == np.ndarray:
+            df_voxel = self._df_in_voxel(self.df, spacing, origin)
```
[the remark this answers is not in the document]
Comment (alishakodibagkar) [reply brainlit/utils/Neuron_trace.py:586]: code coverage no longer an issue

File: docs/notebooks/utils/downloading_brains.ipynb [hunk 4]
```diff
@@ -2,18 +2,9 @@
...
-     "text": [
-      "c:\\users\\shrey\\anaconda3\\envs\\ndd-windows\\lib\\site-packages\\python_jsonschema_objects\\__init__.py:53: UserWarning: Schema version http://json-schema.org/draft-04/schema not recognized. Some keywords and features may not be supported.\n",
-      "  self.schema[\"$schema\"]\n"
-     ]
-    }
-   ],
+   "outputs": [],
    "source": [
     "from brainlit.utils.session import NeuroglancerSession\n",
     "from brainlit.utils.swc import graph_to_paths\n",
```
[the remark this answers is not in the document]
Comment (alishakodibagkar) [reply docs/notebooks/utils/downloading_brains.ipynb:10]: no swc import in this file

[the remark this answers is not in the document]
Comment (tathey1) [reply docs/notebooks/utils/downloading_brains.ipynb:10]: are you sure? - https://github.com/neurodata/brainlit/blob/delete_swc/docs/notebooks/utils/downloading_brains.ipynb

[the remark this answers is not in the document]
Comment (alishakodibagkar) [reply docs/notebooks/utils/downloading_brains.ipynb]: I forgot to commit this change, sorry about that - i just did

**Review by tathey1 — DISMISSED** [review]

**Comment by tathey1** [comment]
Go ahead and fix those merge conflicts and then things will be ready

**Review by tathey1 — APPROVED** [review]
all comments resolved

**Pull request merged**